### PR TITLE
[regression] fix ValidationError arch - Must be one of: x64, Arm64.

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/sut_orchestrator/azure/features.py [hunk 1]
```diff
@@ -1913,10 +1912,13 @@ def enabled(self) -> bool:
 class ArchitectureSettings(schema.FeatureSettings):
     type: str = "Architecture"
     # Architecture in hyper-v
-    arch: str = field(
-        default="x64",
+    arch: str = field(  # type:ignore
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: If so, the type should be consistent with the factory. Refer to `DiskOptionSettings/disk_type` in schema.py.

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/features.py]: BTW, what's the cause behind? I don't see how it got fixed by this change.

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/features.py]: I am still now sure, I just see when issue happened, the setting is `arch {'is_allow_set': False, 'items': ['x64']}`

File: lisa/sut_orchestrator/azure/features.py [hunk 2]
```diff
@@ -1913,10 +1912,13 @@ def enabled(self) -> bool:
 class ArchitectureSettings(schema.FeatureSettings):
     type: str = "Architecture"
     # Architecture in hyper-v
-    arch: str = field(
-        default="x64",
+    arch: Optional[Union[search_space.SetSpace[str], str]] = field(  # type:ignore
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: Keep it as string, and assert req equals cap in `_call_requirement_method`. The reason is that the arch is exclusive each other on both image and vm size. It's impossible to support both in one of them.

File: lisa/sut_orchestrator/azure/features.py [hunk 3]
```diff
@@ -1944,11 +1946,9 @@ def check(self, capability: Any) -> search_space.ResultReason:
...
         ), f"actual: {type(capability)}"
         result = super().check(capability)
-
-        result.merge(
-            search_space.check_setspace(self.arch, capability.arch),
-            "vhd Architecture",
-        )
+        if self.arch != capability.arch:
+            result.result = False
+            result.reasons.append("image arch should be consistent with vm size arch")
```
Comment (LiliDeng) [inline lisa/sut_orchestrator/azure/features.py]: are these changes good?

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/features.py]: this part is ok, please see my previous comments.

File: lisa/sut_orchestrator/azure/features.py [hunk 4]
```diff
@@ -1913,12 +1912,7 @@ def enabled(self) -> bool:
 class ArchitectureSettings(schema.FeatureSettings):
     type: str = "Architecture"
     # Architecture in hyper-v
-    arch: str = field(
-        default="x64",
-        metadata=field_metadata(
-            validate=validate.OneOf(["x64", "Arm64"]),
-        ),
-    )
+    arch: str = "x64"
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: Can we keep the validation? The value should be normalized before set to this field. So it wouldn’t bring other failures later.

File: lisa/sut_orchestrator/azure/features.py [hunk 5]
```diff
@@ -1957,6 +1949,7 @@ def _call_requirement_method(self, method_name: str, capability: Any) -> Any:
             capability, ArchitectureSettings
         ), f"actual: {type(capability)}"
 
+        assert self.arch == capability.arch, "req and capability should be the same"
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: Use aseert_that to print original values

File: lisa/sut_orchestrator/azure/features.py (hunk 4 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/features.py]: I tried with validation, but still encounter this issue. Where should I add the normalized value?

File: lisa/sut_orchestrator/azure/features.py (hunk 5 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/features.py]: updated

File: lisa/sut_orchestrator/azure/features.py (hunk 4 above)
Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/features.py]: Is it possible there is an invalid value? What’s the value before assigned?

File: lisa/sut_orchestrator/azure/features.py [hunk 6]
```diff
@@ -1944,11 +1944,9 @@ def check(self, capability: Any) -> search_space.ResultReason:
...
         ), f"actual: {type(capability)}"
         result = super().check(capability)
-
-        result.merge(
-            search_space.check_setspace(self.arch, capability.arch),
-            "vhd Architecture",
-        )
+        if self.arch != capability.arch:
+            result.result = False
+            result.reasons.append("image arch should be consistent with vm size arch")
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: Log both image and env arch values.

File: lisa/sut_orchestrator/azure/features.py [hunk 7]
```diff
@@ -1957,11 +1955,14 @@ def _call_requirement_method(self, method_name: str, capability: Any) -> Any:
...
+        assert_that(self.arch).described_as(
+            "req and capability should be the same"
+        ).is_equal_to(capability.arch)
+
         value = ArchitectureSettings()
-        if self.arch or capability.arch:
-            value.arch = getattr(search_space, f"{method_name}_setspace_by_priority")(
-                self.arch, capability.arch, ["x64", "Arm64"]
-            )
+        value.arch = next(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: It doesn’t so complicated, just take one of them.

File: lisa/sut_orchestrator/azure/features.py (hunk 6 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/features.py]: updated

File: lisa/sut_orchestrator/azure/features.py (hunk 7 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/features.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**